Sources/SourceControl/GitRepository.swift

```diff
@@ -203,14 +204,31 @@ public class GitRepository: Repository, WorkingCheckout {
     /// The path of the repository on disk.
     public let path: AbsolutePath
 
+    /// The (serial) queue to execute git cli on.
+    private let queue = DispatchQueue(label: "com.apple.swiftpm.gitqueue")
+
     public init(path: AbsolutePath) {
         self.path = path
     }
 
     // MARK: Repository Interface
 
-    public var tags: [String] { return tagsCache.getValue(self) }
-    private var tagsCache = LazyCache(getTags)
+    /// Returns the tags present in repository.
+    public var tags: [String] {
```

Comment:
They are unordered, currently, we could switch them to a set although this is a protocol requirement so that is a larger change.